worksheet.rb

```diff
@@ -1,34 +1,188 @@
-# - Which driver made the most money?
-# - Which driver has the highest average rating?
\ No newline at end of file
+drivers_array = {
+    DR0001: [
+        {
+            date: "Feb 3rd 2016",
+            cost: 10,
+            rider_id: "RD0003",
+            rating: 3
+        },
+        {
+            date: "Feb 3rd 2016",
+            cost: 30,
+            rider_id: "RD0015",
+            rating: 4
+        },
+        {
+            date: "Feb 2nd 2016",
+            cost: 45,
+            rider_id: "RD0003",
+            rating: 2
+        }
+    ],
+    DR0002: [
+        {
+            date: "Feb 2nd 2016",
+            cost: 25,
+            rider_id: "RD0073",
+            rating: 5
+        },
+        {
+            date: "Feb 2 2016",
+            cost: 15,
+            rider_id: "RD0013",
+            rating: 1
+        },
+        {
+            date: "Feb 2nd 2016",
+            cost: 35,
+            rider_id: "RD0066",
+            rating: 3
+        }
+    ],
+    DR0003: [
+        {
+            date: "Feb 2nd 2016",
+            cost: 5,
+            rider_id: "RD0066",
+            rating: 5
+        },
+        {
+            date: "Feb 2nd 2016",
+            cost: 50,
+            rider_id: "RD0003",
+            rating: 2
+        }
+    ],
+    DR0004: [
+        {
+            date: "Feb 2nd 2016",
+            cost: 5,
+            rider_id: "RD0022",
+            rating: 5
+        },
+        {
+            day: "Feb 2nd 2016",
+            cost: 10,
+            rider_id: "RD0022",
+            rating: 4
+        },
+        {
+            date: "Feb 2nd 2016",
+            cost: 20,
+            rider_id: "RD0073",
+            rating: 5
+        }
+    ]
+}
+
+
+# QUESTION ONE: The number of rides each driver has given?
+# -----------------------------------
+# method to see how many rides each driver has done
+def num_of_rides_method(drivers_array)
+  drivers_array.each do |driver_id, rides_array|
+    puts "The driver with ID: #{driver_id} gave #{rides_array.length} rides"
+  end
+end
+
+num_of_rides_method(drivers_array)
+
+# QUESTION TWO: The total amount of money each driver has made?
+# -----------------------------------
+# method that sees how much money each driver made
+
+def driver_earning_method(driver_array)
+  drivers_earnings= 0
+  driver_array.each do |driver,ride_array|
+    total = 0
+    ride_array.each do |ride|
+      total += ride[:cost]
+    end
+    puts "The driver with ID: #{driver} made $ #{total}"
+  end
+end
+puts ""
+
+driver_earning_method(drivers_array)
+puts ""
+
+
+#QUESTION THREE # - What is the average rating for each driver?
+
+drivers_array.each do |driver,rides_array|
+  total_driver_ratings = 0
+
+  rides_array.each do |ride_hash|
+    total_driver_ratings += ride_hash[:rating]
+  end
+  average= total_driver_ratings.to_f/rides_array.length
+  puts "Driver with ID: #{driver}, has a rating average of #{average.round(1)}"
+end
+puts ""
```

Comment:
Consider encapsulating this functionality into a method to be consistent with the work for questions 1 and 2.

Also, you might consider combining the work for question 1-3 into a single method since they all require the same driving each loop. It is a tricky balance between dividing up functionality so make each method truly only do one thing, and reducing repeated code.